hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/BaseScheduleCompactionActionExecutor.java

```diff
@@ -63,7 +63,7 @@ public BaseScheduleCompactionActionExecutor(HoodieEngineContext context,
                   + ", Compaction scheduled at " + instantTime));
       // Committed and pending compaction instants should have strictly lower timestamps
       List<HoodieInstant> conflictingInstants = table.getActiveTimeline()
-          .getWriteTimeline().getInstants()
+          .getWriteTimeline().filterCompletedAndCompactionInstants().getInstants()
```

Comment:
@swuferhong @danny0405 If you take a look at the previous version of this file, the method is called before was `commitsAndCompactionTimeline` -> https://github.com/apache/hudi/blob/3e71c915271d77c7306ca0325b212f71ce723fc0/hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieDefaultTimeline.java#L110

This follows the same behavior as getWriteTimeline().getInstants(). Can you please explain what is a possible bug here ?

Reply:
No, the method `getWriteTimeline()` does not really follow the behavior of `filterCompletedAndCompactionInstants`,
`getWriteTimeline()` actually may include any `INFLIGHT` instants but `filterCompletedAndCompactionInstants` only include `COMPACTION` `INFLIGHT` instants.

We should allow scheduling compaction if there are inflight commits or inflight delta_commits.

Reply:
You are right, I looked at the wrong API. This is the correct one -> https://github.com/apache/hudi/blob/3e71c915271d77c7306ca0325b212f71ce723fc0/hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieDefaultTimeline.java#L104.

Reply:
Take a look at the comments
```java
// Committed and pending compaction instants should have strictly lower timestamps
```
I think the code before that used `commitsAndCompactionTimeline()` is already wrong, it add restrictions that we can not generate compaction plan when there are inflight commits, of course we can actually.

Reply:
Okay got it. So you're saying this bug was always there even in 0.7 release ? Can we please add a test case for this before landing ?

Reply:
Sure, let us add a test case.